account_invoice_pricelist/model/account_invoice.py

```diff
@@ -0,0 +1,58 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Account - Pricelist on Invoices for Odoo
+#    Copyright (C) 2015-Today GRAP (http://www.grap.coop)
+#    @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    # Column Section
+    pricelist_id = fields.Many2one(
+        comodel_name='product.pricelist', string='Pricelist',
+        help="The pricelist of the partner, when the invoice is created"
+        " or the partner has changed. This is a technical field used"
+        " for reporting.")
+
+    @api.multi
+    def onchange_partner_id(
```

Comment:
@moylop260 : 
I'm not sure you're pointing a stable version and with the same framework.
in V8 (last stable version) : 
account module: product_id_change : https://github.com/odoo/odoo/blob/8.0/addons/account/account_invoice.py#L1308
account_asset : (no overload) : https://github.com/odoo/odoo/blob/8.0/addons/account_asset/account_asset_invoice.py

Do you have some example in V8 ?
kind regards.